Dockerfile

```diff
@@ -55,6 +55,7 @@ COPY php.ini /etc/php/7.4/cli/conf.d/99-docker.ini
 
 RUN ["ln", "-vs", "/usr/share/icingaweb2/bin/icingacli", "/usr/local/bin/"]
 RUN ["icingacli", "setup", "config", "webserver", "apache", "--path=/", "--file=/etc/apache2/conf-enabled/icingaweb2.conf"]
+RUN echo 'SetEnvIf X-REMOTE-USER "(.*)" REMOTE_USER=$0' > /etc/apache2/conf-enabled/allow-remote-user.conf
```

Comment:
Well, as long as end user headers X-REMOTE-USER are properly filtered away by the reverse proxy... 👍

Reply:
What happens if it isn't filtered? Would that be a problem? If so, doing this by default would be a bad idea.

Reply:
You need to explicitly set icingaweb to use external users (`icingaweb.authentication.autologin.backend: external`)  since only the ExternalBackend class is using that Variable. 

At that point you need to populate the variable because this is the only way to tell icingaweb the username: https://github.com/Icinga/icingaweb2/blob/main/library/Icinga/Authentication/User/ExternalBackend.php#L70

==> if you are not using an external backend, the var is unused. If you are using one, you are setting it in your authentication flow.

Reply:
_setting_ is the magic word here. If, in contrast, you _add_ a header to the user-supplied ones, you'll have a problem. 😅

Reply:
I haven't used apache in a while, but we could also think about e.g.:

if the docker container has environment variable `ICINGA_REMOTE_USER_HEADER` set, use the value of the variable as the `REMOTE_USER` header.

If i read the apache documentation correctly, this could be done with the  mod_env  PassEnv directive and a few if statements.


I need to see when i have time to check if that solution would work.